python/lsst/pipe/base/graph/graph.py

```diff
@@ -613,12 +628,13 @@ def subset(self: _T, nodes: Union[QuantumNode, Iterable[QuantumNode]]) -> _T:
         # convert to standard dict to prevent accidental key insertion
         quantumDict: Dict[TaskDef, Set[Quantum]] = dict(quantumMap.items())
         # Create an empty graph, and then populate it with custom mapping
-        newInst = type(self)({})
+        newInst = type(self)({}, universe=self._universe)
         newInst._buildGraphs(
             quantumDict,
             _quantumToNodeId={n.quantum: n.nodeId for n in nodes},
             _buildId=self._buildId,
             metadata=self._metadata,
+            universe=self._universe,
```

Comment:
May be perfectly correct, but seems strange to create the object with the universe value on line 631 and then pass in the same value to that object's method on line 637.

Reply:
I agree it's strange; it's an artifact of some hoops the code jumps through to avoid having the "usual" `__init__` logic do a lot of unnecessary work.  Basically, passing the universe to `__init__` just saves `__init__` from complaining that it doesn't have a universe, and then the next call actually sets up the graph (including setting the universe).